sql: add roach test / workload for testing connection latencies

RichardJCai opened the pull request:
Add roach test and workload test for testing connection latencies.

Release note: None

Making it a private test right now (not available through `cockroach workload`)

Need to also add it to teamcity and make sure the results are reported in roachperf, will open PR there.

Resolves https://github.com/cockroachdb/cockroach/issues/59394

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62166)
<!-- Reviewable:end -->

rafiss commented on pkg/workload/connectionlatency/connectionlatency.go:
```diff
@@ -0,0 +1,97 @@
+	"github.com/cockroachdb/cockroach/pkg/workload"
+	"github.com/cockroachdb/cockroach/pkg/workload/histogram"
+	"github.com/cockroachdb/errors"
+	"github.com/jackc/pgx"
```
let's use `github.com/jackc/pgx/v4`

rafiss commented on pkg/workload/connectionlatency/connectionlatency.go:
```diff
@@ -0,0 +1,97 @@
+		return err
+	}
+	start := timeutil.Now()
+	conn, err := pgx.Connect(connCfg)
```
i think we'd want a `defer conn.Close()` otherwise the connection will remain open forever? (i think -- curious to hear what you see when trying this)

rafiss commented on pkg/workload/connectionlatency/connectionlatency.go:
```diff
@@ -0,0 +1,97 @@
+	ctx context.Context, urls []string, reg *histogram.Registry,
+) (workload.QueryLoad, error) {
+	ql := workload.QueryLoad{}
+	if len(urls) != 1 {
```
is this always 1? it would be nice to be able to run this test in different configurations -- single-node, multi-node in 1 region, and multi-region. that way we can compare how it differs for all these.

take a look at some of the other tests on https://roachperf.crdb.dev/
some of them have `nodes=X` and `multiregion` specifiers. not sure how to get that type of configuration, but i think we want it here.

rafiss reviewed (changes requested):
glad this wasn't too bad! just some minor comments

RichardJCai reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62166)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[pkg/workload/connectionlatency/connectionlatency.go, line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MW0t0nBx1soSF3KMGH7-r1-20:-MW5zc4ICOG8FAlJVwFM:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b5604bebd60097df6b8d45d3bc63f94a410179db/pkg/workload/connectionlatency/connectionlatency.go#L20)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

let's use `github.com/jackc/pgx/v4`
</blockquote></details>

Done.
___
*[pkg/workload/connectionlatency/connectionlatency.go, line 60 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MW0t0nBx1soSF3KMGH7-r1-60:-MW5zcV-5ct65U3zyVwk:b473izz) ([raw file](https://github.com/cockroachdb/cockroach/blob/b5604bebd60097df6b8d45d3bc63f94a410179db/pkg/workload/connectionlatency/connectionlatency.go#L60)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

is this always 1? it would be nice to be able to run this test in different configurations -- single-node, multi-node in 1 region, and multi-region. that way we can compare how it differs for all these.

take a look at some of the other tests on https://roachperf.crdb.dev/
some of them have `nodes=X` and `multiregion` specifiers. not sure how to get that type of configuration, but i think we want it here.
</blockquote></details>

Changed it to 1,3,5 nodes and one multiregion test with 6 nodes. Let me know what you think
___
*[pkg/workload/connectionlatency/connectionlatency.go, line 82 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MW0t0nBx1soSF3KMGH7-r1-82:-MW5zhHl5xbqn-Bp6XsS:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b5604bebd60097df6b8d45d3bc63f94a410179db/pkg/workload/connectionlatency/connectionlatency.go#L82)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i think we'd want a `defer conn.Close()` otherwise the connection will remain open forever? (i think -- curious to hear what you see when trying this)
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

rafiss reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62166)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @RichardJCai)

___
*[pkg/cmd/roachtest/connection_latency.go, line 20 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWAkbdQ1jfPXb94_0yb:-MWAkbdQ1jfPXb94_0yc:bjwmj8k) ([raw file](https://github.com/cockroachdb/cockroach/blob/7bdefb529530f9f0139a99f6ad899f2d457ca00b/pkg/cmd/roachtest/connection_latency.go#L20)):*
> ```Go
> )
> 
> func registerConnectionLatencyTest(r *testRegistry) {
> ```

oh one more thing i thought of! does this test both `insecure` and `secure` clusters? we've seen that connection behaves differently when password authn is needed for non-root users

or if that seems like too many tests, we really only need to test with `secure` and make sure not to connect with `root`
___
*[pkg/cmd/roachtest/connection_latency.go, line 39 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWAlcyN7ECrU64Ar8os:-MWAlcyO3CiUQVH2Fvo_:b-kql7n) ([raw file](https://github.com/cockroachdb/cockroach/blob/7bdefb529530f9f0139a99f6ad899f2d457ca00b/pkg/cmd/roachtest/connection_latency.go#L39)):*
> ```Go
> 
> 	geoZones := []string{"us-east1-b", "us-west1-b", "europe-west2-b"}
> 	if cloud == aws {
> ```

i think it's fine if the test only works in GCE
___
*[pkg/cmd/roachtest/connection_latency.go, line 58 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWAli5Q4Wb_gdSvzwsB:-MWAli5Q4Wb_gdSvzwsC:b-x6o774) ([raw file](https://github.com/cockroachdb/cockroach/blob/7bdefb529530f9f0139a99f6ad899f2d457ca00b/pkg/cmd/roachtest/connection_latency.go#L58)):*
> ```Go
> 
> 	// Copying over multiregion configuration from indexes.go
> 	numMultiRegionNodes := 6
> ```

i think it's more common for there to be 3 nodes per per region, so we'd want something more like what the tpcc test does (9 nodes)


<!-- Sent from Reviewable.io -->

RichardJCai committed:
roachtest: add --secure and --user flags to workload command

Support running workload with secure mode (sslmode=require) and
allow a user to be passed in.

Previously, roachtests were only run as root on insecure mode leaving
some gaps in our testing.

Release note: None

RichardJCai reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62166)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[pkg/cmd/roachtest/connection_latency.go, line 20 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWAkbdQ1jfPXb94_0yb:-MWUzD4KBenaUuvZja4F:b-fk1k02) ([raw file](https://github.com/cockroachdb/cockroach/blob/7bdefb529530f9f0139a99f6ad899f2d457ca00b/pkg/cmd/roachtest/connection_latency.go#L20)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

oh one more thing i thought of! does this test both `insecure` and `secure` clusters? we've seen that connection behaves differently when password authn is needed for non-root users

or if that seems like too many tests, we really only need to test with `secure` and make sure not to connect with `root`
</blockquote></details>

Made the test only use secure & not root
___
*[pkg/cmd/roachtest/connection_latency.go, line 39 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWAlcyN7ECrU64Ar8os:-MWUzFVb0qPwBh74bSzm:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/7bdefb529530f9f0139a99f6ad899f2d457ca00b/pkg/cmd/roachtest/connection_latency.go#L39)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i think it's fine if the test only works in GCE
</blockquote></details>

Done.
___
*[pkg/cmd/roachtest/connection_latency.go, line 58 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWAli5Q4Wb_gdSvzwsB:-MWUzG2t6RuP8lIYCKDh:bs4eab1) ([raw file](https://github.com/cockroachdb/cockroach/blob/7bdefb529530f9f0139a99f6ad899f2d457ca00b/pkg/cmd/roachtest/connection_latency.go#L58)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i think it's more common for there to be 3 nodes per per region, so we'd want something more like what the tpcc test does (9 nodes)
</blockquote></details>

Did 9 


<!-- Sent from Reviewable.io -->

rafiss reviewed (commented):
just have questions that we can discuss!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-:-MWWKy3Q6eppkrxYAx3z:b-dvs5zu)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @RichardJCai)

___
*[pkg/cmd/roachtest/connection_latency.go, line 42 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWWKQO95smhjelbC2dA:-MWWKQO95smhjelbC2dB:bh0dnaq) ([raw file](https://github.com/cockroachdb/cockroach/blob/be049c571c62900f820bba673dedbf6712784f15/pkg/cmd/roachtest/connection_latency.go#L42)):*
> ```Go
> 
> 		workloadCmd := fmt.Sprintf(
> 			`./workload run connectionlatency --user testuser --secure --duration 30s --histograms=%s/stats.json`,
> ```

would we want to pass `--urls` as a flag here? again, just asking since i'm not sure how it all works
___
*[pkg/cmd/roachtest/connection_latency.go, line 51 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWWKWcREJGn2RdE89gY:-MWWKWcREJGn2RdE89gZ:b-jmcfcq) ([raw file](https://github.com/cockroachdb/cockroach/blob/be049c571c62900f820bba673dedbf6712784f15/pkg/cmd/roachtest/connection_latency.go#L51)):*
> ```Go
> 	geoZonesStr := strings.Join(geoZones, ",")
> 
> 	nodesConfig := []int{1, 3, 5}
> ```

maybe we just need to test the 3-node cluster and the 9-node multiregion cluster. well, the only reason i'm saying this is just to avoid extra cruft, and since i don't have a good sense of how much overhead the additional setup is. if all the extra tests are cheap, then fine to leave as is
___
*[pkg/workload/cli/run.go, line 194 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWWJyuM0RS_Ic90Wkou:-MWWJyuM0RS_Ic90Wkov:bmsyh7f) ([raw file](https://github.com/cockroachdb/cockroach/blob/be049c571c62900f820bba673dedbf6712784f15/pkg/workload/cli/run.go#L194)):*
> ```Go
> 
> 		if len(urls) == 0 {
> 			crdbDefaultURL := fmt.Sprintf(`postgres://%s@localhost:26257?sslmode=disable`, *user)
> ```

i'm not familiar with how this works -- why does it connect to localhost? where does it connect from?
___
*[pkg/workload/connectionlatency/connectionlatency.go, line 60 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWWK6coE_sHiTUimFu3:-MWWK6coE_sHiTUimFu4:biors8f) ([raw file](https://github.com/cockroachdb/cockroach/blob/be049c571c62900f820bba673dedbf6712784f15/pkg/workload/connectionlatency/connectionlatency.go#L60)):*
> ```Go
> ) (workload.QueryLoad, error) {
> 	ql := workload.QueryLoad{}
> 	if len(urls) != 1 {
> ```

do we expect the urls length to be 1 even for the multi-node clusters?


<!-- Sent from Reviewable.io -->

RichardJCai reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62166)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @RichardJCai)

___
*[pkg/cmd/roachtest/connection_latency.go, line 42 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWWKQO95smhjelbC2dA:-MW_AYD-EajWxvgBH3Db:b-lbktqr) ([raw file](https://github.com/cockroachdb/cockroach/blob/be049c571c62900f820bba673dedbf6712784f15/pkg/cmd/roachtest/connection_latency.go#L42)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

would we want to pass `--urls` as a flag here? again, just asking since i'm not sure how it all works
</blockquote></details>

That's an option if we want to make it more configurable, by default the workload runs on the same machine as the crdb node so localhost works.
___
*[pkg/workload/connectionlatency/connectionlatency.go, line 60 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWWK6coE_sHiTUimFu3:-MW_AfsKEk4LXD0hZBdH:b-pkh0ew) ([raw file](https://github.com/cockroachdb/cockroach/blob/be049c571c62900f820bba673dedbf6712784f15/pkg/workload/connectionlatency/connectionlatency.go#L60)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

do we expect the urls length to be 1 even for the multi-node clusters?
</blockquote></details>

Yeah if we don't specify a url, it'll use the localhost one, so the workload will connect using the node on the same machine. We can change this though


<!-- Sent from Reviewable.io -->

rafiss reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62166)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @RichardJCai)

___
*[pkg/cmd/roachtest/connection_latency.go, line 42 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWWKQO95smhjelbC2dA:-MWaHAjm-VrZn_CdZtc7:bo9nzcr) ([raw file](https://github.com/cockroachdb/cockroach/blob/be049c571c62900f820bba673dedbf6712784f15/pkg/cmd/roachtest/connection_latency.go#L42)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

That's an option if we want to make it more configurable, by default the workload runs on the same machine as the crdb node so localhost works.
</blockquote></details>

which machine does `workload` run on if it's a multi-node cluster?

either way, for a multi-node cluster, i think we'd want a broad sample by testing connections to all the nodes in the cluster. this would show us issues like the one where we saw that it had to fetch data across different regions.


<!-- Sent from Reviewable.io -->

RichardJCai committed:
sql: add roach test / workload for testing connection latencies

Add roach test and workload test for testing connection latencies.

Release note: None

RichardJCai reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62166)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[pkg/cmd/roachtest/connection_latency.go, line 42 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWWKQO95smhjelbC2dA:-MWfQ4KVD7mHyVyTCFKm:b-7tbrc7) ([raw file](https://github.com/cockroachdb/cockroach/blob/be049c571c62900f820bba673dedbf6712784f15/pkg/cmd/roachtest/connection_latency.go#L42)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

which machine does `workload` run on if it's a multi-node cluster?

either way, for a multi-node cluster, i think we'd want a broad sample by testing connections to all the nodes in the cluster. this would show us issues like the one where we saw that it had to fetch data across different regions.
</blockquote></details>

Updated so each node will connect to each other node. 

I wonder if we should make the tracking a bit more granular for this though, ie region to region latency test.
___
*[pkg/cmd/roachtest/connection_latency.go, line 51 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWWKWcREJGn2RdE89gY:-MWfQCeL9znYLze5JRNK:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/be049c571c62900f820bba673dedbf6712784f15/pkg/cmd/roachtest/connection_latency.go#L51)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

maybe we just need to test the 3-node cluster and the 9-node multiregion cluster. well, the only reason i'm saying this is just to avoid extra cruft, and since i don't have a good sense of how much overhead the additional setup is. if all the extra tests are cheap, then fine to leave as is
</blockquote></details>

Done.
___
*[pkg/workload/cli/run.go, line 194 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MWWJyuM0RS_Ic90Wkou:-MWfQDcs0QFbQ-B1PO9P:b-l7cfwn) ([raw file](https://github.com/cockroachdb/cockroach/blob/be049c571c62900f820bba673dedbf6712784f15/pkg/workload/cli/run.go#L194)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i'm not familiar with how this works -- why does it connect to localhost? where does it connect from?
</blockquote></details>

This is if no URL is specified, we use localhost by default. I believe it's assumed the VM will host the node.


<!-- Sent from Reviewable.io -->

RichardJCai commented:
> which machine does workload run on if it's a multi-node cluster?
>
> either way, for a multi-node cluster, i think we'd want a broad sample by testing connections to all the nodes in the cluster. this would show us issues like the one where we saw that it had to fetch data across different regions.

@rafiss following up more on this point, I mentioned `I wonder if we should make the tracking a bit more granular for this though, ie region to region latency test.` is there a way to check which region a node is on? I couldn't find a way right now so the latencies tracked can vary depending on which node is connecting to which. Do you think this is sufficient?

I've attached the output and some of the p99(ms) latencies are really bad, ~800ms for connect.
[run_202535.430_n1-9_workload_run_connectionlatency.log](https://github.com/cockroachdb/cockroach/files/6214290/run_202535.430_n1-9_workload_run_connectionlatency.log)

rafiss commented:
I think at the moment, a p99 of 800ms might be expected. I think the auth code makes either 2 or 3 KV roundtrips, so if all those are cross-region, this could happen. See these related issues https://github.com/cockroachdb/cockroach/issues/58869 https://github.com/cockroachdb/cockroach/issues/36160

But yeah I agree it would be nice to be able to track this granularly. I think `SHOW LOCALITY` might give you what you want to find the node's region: https://www.cockroachlabs.com/docs/stable/show-locality.html

The other issue with this though is that the leaseholder placements won't be fixed across different test runs. So e.g. uswest1 could be slow one night, then the next night useast1 could be slow.

Maybe the test should do an `ALTER RANGE system CONFIGURE ZONE` to pin the system leaseholder to the same region in each test? IDK if it's super needed though. it might just be fine that different regions are slow across runs. let's start by not changing any zone configs and if the test is too hard to understand we can add that later.

RichardJCai reviewed (commented):
Updated the output to show which region is used. It assumes that the regions/zones stay consistent throughout the test's history.

```_
elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            397           13.2    223.5    226.5    234.9    234.9    251.7  connect-to-cloud=gce,region=europe-west2,zone=europe-west2-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            150            5.0    427.9    436.2    436.2    436.2    453.0  connect-to-cloud=gce,region=us-east1,zone=us-east1-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            108            3.6    587.2    604.0    604.0    604.0    604.0  connect-to-cloud=gce,region=us-west1,zone=us-west1-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            653           21.8    370.1    226.5    738.2    738.2    738.2  select
```

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-:-MX59gGtCXkeffIjzwdp:b-opzv1b)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)



<!-- Sent from Reviewable.io -->

RichardJCai commented:
Okay I think I finally have this in a state I'm satisfied with.

Example output

Latencies connecting from us-west
```
_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            154            5.1    519.1    520.1    536.9    570.4    570.4  connect-from-us-west1-b-to-cloud=gce,region=europe-west2,zone=europe-west2-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            208            6.9    444.6    453.0    453.0    453.0    453.0  connect-from-us-west1-b-to-cloud=gce,region=us-east1,zone=us-east1-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            445           14.8    267.4    268.4    285.2    285.2    302.0  connect-from-us-west1-b-to-cloud=gce,region=us-west1,zone=us-west1-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            805           26.8    404.9    285.2    671.1    671.1    704.6  select
```

Latencies connecting from us-east
```
_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            192            6.4    447.3    453.0    469.8    536.9    570.4  connect-from-us-east1-b-to-cloud=gce,region=europe-west2,zone=europe-west2-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0           1194           39.8     99.2    100.7    104.9    109.1    318.8  connect-from-us-east1-b-to-cloud=gce,region=us-east1,zone=us-east1-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            200            6.7    464.1    469.8    486.5    536.9    536.9  connect-from-us-east1-b-to-cloud=gce,region=us-west1,zone=us-west1-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0           1586           52.9    207.0    100.7    536.9    570.4    637.5  select
```

Latencies connecting from eu-west
```
_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0           8340          278.0     13.2     13.1     18.9     22.0     33.6  connect-from-europe-west2-b-to-cloud=gce,region=europe-west2,zone=europe-west2-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            168            5.6    541.7    570.4    570.4    570.4    570.4  connect-from-europe-west2-b-to-cloud=gce,region=us-east1,zone=us-east1-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0            116            3.9    785.6    805.3    805.3    838.9    838.9  connect-from-europe-west2-b-to-cloud=gce,region=us-west1,zone=us-west1-b

_elapsed___errors_____ops(total)___ops/sec(cum)__avg(ms)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)__total
   30.0s        0           8622          287.4     37.6     14.2     21.0    906.0    973.1  select
```

rafiss reviewed (approved):
cool!! this looks great

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-:-MXD67Oy1BWX6AToXhIT:b6brpqp)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @RichardJCai)

___
*[pkg/workload/connectionlatency/connectionlatency.go, line 82 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/62166#-MXD5XsR2ZEDy6ZOD0OM:-MXD5XsR2ZEDy6ZOD0ON:b-2oh3w0) ([raw file](https://github.com/cockroachdb/cockroach/blob/3a6415a97b1a88b3bb9e48f40a7f533ff6df5fb9/pkg/workload/connectionlatency/connectionlatency.go#L82)):*
> ```Go
> 
> 		var locality string
> 		err = conn.QueryRow(ctx, "SHOW LOCALITY").Scan(&locality)
> ```

one thing i just realized is that it is expected for the latency to be high when connection from one region to a separate region, and there isn't much we can do about that

so the most important metrics out of this test will be the `useast->useast`, `uswest->uswest`, and `euwest->euwest` connection latencies. the other ones may not be as important, but i think can still be useful so we can compare.

anyway, just noting this down so we have a better idea of how we'll use this test in the future.


<!-- Sent from Reviewable.io -->

RichardJCai commented:
Thanks for the review!!

bors r=rafiss
